### PR TITLE
HomeScreen/Header

### PR DESCRIPTION
**Pull request by saurabhalp**
created header component takes `username` and `userProfile`as input.

### CODE REVIEW

File: src/UI/Components/home_header/index.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+
+import { View, Text, Image, StyleSheet, ImageSourcePropType } from 'react-native';
+import AvatarCircle from '../AvatarCircle';
+import { styles } from './styles';
+
+interface HeaderProps {
```
Comment (Jha2508) [inline src/UI/Components/home_header/index.tsx]: why are we using `interface`? Please read once why `interface` is used. use `type` here instead, and add docs here

File: src/UI/Components/home_header/index.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  username: string;
+  profileImage: ImageSourcePropType;
+}
+
+const Header: React.FC<HeaderProps> = ({ username, profileImage }) => {
```
Comment (Jha2508) [inline src/UI/Components/home_header/index.tsx]: We do not use React.FC component at all, just make an arrow function taking your type as props 
for eg 
`const Header = (props: HeaderProps)=>{...}`

File: src/UI/Components/home_header/index.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  return (
+    <View style={styles.header}>
+        <View style={styles.subHeader}>
+        <Text style={styles.welcomeText}>Welcome Back</Text>
+        <Text style={styles.username}>{username}</Text>
+        </View>
+   
+      <View>  
+      <AvatarCircle radius={20} imageSource={profileImage}
```
Comment (Jha2508) [inline src/UI/Components/home_header/index.tsx]: why radius is hardcoded? take it in props

File: src/UI/Components/home_header/index.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+interface HeaderProps {
+  username: string;
+  profileImage: ImageSourcePropType;
+}
+
+const Header: React.FC<HeaderProps> = ({ username, profileImage }) => {
+  return (
+    <View style={styles.header}>
+        <View style={styles.subHeader}>
+        <Text style={styles.welcomeText}>Welcome Back</Text>
```
Comment (Jha2508) [inline src/UI/Components/home_header/index.tsx]: Move `Welcome Back` to constants

**Comment by saurabhalp** [comment]
-Replaced interface with type
-added docs
-Updated AvatarCircle interface with type
-Moved Welcome Back to Constant

File: src/UI/Components/AvatarCircle/index.tsx [hunk 5]
```diff
@@ -1,11 +1,17 @@
...
-    imageSource: ImageSourcePropType; // The image source, which can be a local image or a URL
-  }
-const AvatarCircle: React.FC<CircleProps> = ({ radius, imageSource }) => {
+import React from 'react';
+/**
+ * AvatarCircleProps defines the properties required for the Header component.
+ * @property {ImageSourcePropType} profileImage - The source of the image.
+ * @property {number} Radius - The radius of the avatar circle.
+ */
+type AvatarCircleProps = {
```
Comment (Jha2508) [inline src/UI/Components/AvatarCircle/index.tsx:10]: The types should be below the declaration of the UI COMPONENT

File: src/UI/Components/home_header/index.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  username: string;
+  profileImage: ImageSourcePropType;
+  avatarRadius: number;
+};
+
+
+/**
+ * Header component displays a welcome message, username, and user avatar.
+ */
+const Header = ({ username, profileImage, avatarRadius }: HeaderProps) => {
```
Comment (Jha2508) [inline src/UI/Components/home_header/index.tsx]: Use 'props : HeaderProps' here and access properties using props.propName. Do this thing in all components

File: src/UI/Components/home_header/index.tsx
Comment (Aashirvaad-P) [inline src/UI/Components/home_header/index.tsx:1]: Do focus on naming of files.

**Review by Jha2508 — APPROVED** [review]

**Pull request merged**